docs/example.md

```diff
@@ -153,7 +153,7 @@ import (
 
 func main() {
   server := &haberdasherserver.Server{} // implements Haberdasher interface
-  twirpHandler := haberdasher.NewHaberdasherServer(server, nil)
+  twirpHandler := haberdasher.NewHaberdasherServer(server)
```

Comment:
Using `nil` still works (for backwards compatibility), but the last constructor argument is now a variadic `opts ...interface{}`, which could be `nil`, the `hooks` or an option `WithHooks(hooks)`. I removed the leading `nil` value from examples and tests where hooks are not part of the focus.